Did some changes to the POM and GRADLE file as the required libraries di...

sambitgaan opened the pull request:
Fixes #178 

Did some changes to the POM and GRADLE file as the required libraries did not load. Also added the libraries to the eclipse library so that ecplise project should not complain about missing libraries.

cogmission commented:
@sambitgaan 
Thank you for looking into this. I was going to resolve this too, but I was also trying to combine it with other fixes I'm working on. 

We can go ahead and submit fixes for this right away, but I have a couple of requests. 
1.) Please change the Maven pom implementation to:  (replace log4j, keep in the slf4j config)
<dependency> 
  <groupId>ch.qos.logback</groupId>
  <artifactId>logback-classic</artifactId>
  <version>1.0.13</version>
</dependency>

Logback is log4j The-Next-Generation. :)

2.) In the gradle file, please remove:
compile group: 'org.slf4j', name: 'slf4j-log4j12', version:'1.7.10'
-and insert-
compile group: 'ch.qos.logback', name: 'logback-classic', version: '1.+'

Also, Please remember that PR's now require an associated "issue" to be created and linked to. I will create an issue for this, then edit your original comment to include "fixes #blah"  - so that the validator will pick it up.

cogmission commented:
Oh yes. Please add  logback-core-1.0.13.jar, and logback-classic-1.0.13.jar to the "libs" directory and remove the log4j jar.

cogmission commented:
@sambitgaan 
I just realized that you might not understand what slf4j is?  Slf4j is an interface only. It ascribes to the implementation interfaces of log4j, java.util.logging etc. - meaning that when you use the slf4j interface it allows the *end user* to use whatever logging system they want. By default, slf4j uses a "no-logger" implementations and messages simply aren't logged. It should now interrupt actual program flow at all. Slf4j is able to use whatever logging framework happens to be on the end user's class path, and messages will simply be output to that logging library.

I suggested putting in logback's two dependencies in the "libs" directory for convenience because "logback" is log4j's upgrade. However, the beauty of slf4j is that no logging implementation **must** be provided - the user can opt out of using one altogether!

In addition, we have the test configuration of logback so that we can see output in a "test" context (when running tests for instance) - because output is desired at all times for tests. By only putting it in the test environment we keep it out of the "production" execution pipeline.

You may get a message output on standard err saying that no logging configuration has been provided and slf4j is defaulting to its "no-operation" config. This is ok, and expected. However, all tests should pass. 

**EDITED:** 
The tests all pass so the above request is rescinded. Please simply put the logback-core-1.0.13.jar, and logback-classic-1.0.13.jar files in the "libs" directory for convenience and roll back the pom file and the build.gradle files.

cogmission commented:
Closing this, as it doesn't really address an actual issue.

sambitgaan commented:
Thanks David for the information regarding Slf4j functionality or the
reason why it is being used.
Also I have never used Slf4j in my earlier projects, hence don't know much
of it's functionality.
Anyway as I know now so I will try to change it in my code base accordingly
and will try to use the logback library.
Will rise a hand in case I face any difficulty.


Regards,
Sambit



On Sat, Feb 7, 2015 at 11:33 PM, David Ray <notifications@github.com> wrote:

> Closing this, as it doesn't really address an actual issue.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/numenta/htm.java/pull/179#issuecomment-73375202>.
>

sambitgaan commented:
As suggested I have reverted the changes made to the Gradle config and POM
file.
Also I have added the logback library in the test scope.
This runs find on my system and all the tests are running including the
test shared separately on mail by Peter Dove.
Actually I faced the missing log library issue when I started running the
code submitted by Peter Dove.
Any way the changes are committed to my branch. Please review it and it
appropriate the merge it with master branch.

Regards,
Sambit Gaan


On Sun, Feb 8, 2015 at 12:03 PM, Sambit Gaan <sambit.gaan@gmail.com> wrote:

> Thanks David for the information regarding Slf4j functionality or the
> reason why it is being used.
> Also I have never used Slf4j in my earlier projects, hence don't know much
> of it's functionality.
> Anyway as I know now so I will try to change it in my code base
> accordingly and will try to use the logback library.
> Will rise a hand in case I face any difficulty.
>
>
> Regards,
> Sambit
>
>
>
> On Sat, Feb 7, 2015 at 11:33 PM, David Ray <notifications@github.com>
> wrote:
>
>> Closing this, as it doesn't really address an actual issue.
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/numenta/htm.java/pull/179#issuecomment-73375202>.
>>
>
>
